rdtools/soiling.py

```diff
@@ -343,6 +351,17 @@ def _calc_monte(self, monte, method='half_norm_clean'):
               the interval.
         '''
 
+        # Raise a warning if there is >20% invalid data
+        if (method == 'half_norm_clean') or (method == 'random_clean'):
+            valid_fraction = self.analyzed_daily_df['valid'].sum()/len(self.analyzed_daily_df)
+            if valid_fraction <= 0.8:
+                warnings.warn('20% or more of the daily data is assigned to invalid soiling '
+                              'intervals. This can be problematic with the "half_norm_clean" '
+                              'and "random_clean" cleaning assumptions. Consider more permissive '
+                              'validity criteria such as increasing "max_relative_slope_error" '
+                              'and/or "max_negative_step" and/or decreasing "min_interval_length".'
+                              ' Alternatively, consider using method="perfect_clean".'
```

Comment:
Is this in reference to #234?  Might be good to add that breadcrumb in case folks are interesting in what exactly is problematic.

Also, should the 20% threshold be exposed as an optional parameter?

Reply:
It's more related to the retention of invalid intervals at the start and end. I will open an issue on that to leave a picture for folks.

Reply:
Since it's only a warning, I don't think we need to expose another parameter. If someone is in the weeds enough to want to change that, they'd easily be able to detect whatever fraction they are worried about. This is more for folks who wouldn't have it on their radar.